packages/backend-common/src/service/lib/ServiceBuilderImpl.ts

```diff
@@ -120,34 +154,68 @@ export class ServiceBuilderImpl implements ServiceBuilder {
         reject(e);
       });
 
-      const server = stoppable(
-        app.listen(port, host, () => {
-          logger.info(`Listening on ${host}:${port}`);
-        }),
-        0,
-      );
+      const useHttps = baseUrl.indexOf(DEFAULT_PROTOCOL) > -1;
+      let server: http.Server;
+
+      if (useHttps) {
```

Comment:
We need to determine whether to use HTTPS based on the presence of config instead. You'd configure a `https://` `baseUrl` even if you're behind a proxy that does TLS termination for you, and then you wouldn't want to serve HTTPS.

Reply:
Hmmm as I see it people will look at the baseUrl and go mkay the backend is https based. A second configuration value only serves to make our config more bloated and it will open up for scenarios where people can actually end up configuring a https server that mounts a http URL for the rest of the system to communicate through. This will probably cause a lot of funky CORS bugs down the line?

Reply:
With a proxy in front that terminates TLS you'll want the backend to serve HTTP, while the baseUrl should be HTTPS.